vendor/brave-ios/Ledger/BATBraveLedger.mm

```diff
@@ -550,103 +553,124 @@ - (void)tipPublisherDirectly:(BATPublisherInfo *)publisher amount:(double)amount
+    for (BATPromotion *promotion in promos) {
+      if (promotion.status == BATPromotionStatusFinished) {
+        [self.mFinishedPromotions addObject:promotion];
+      } else if (promotion.status == BATPromotionStatusActive ||
+                 promotion.status == BATPromotionStatusAttested) {
+        [self.mPendingPromotions addObject:promotion];
+        bool isUGP = promotion.type == BATPromotionTypeUgp;
+        auto notificationKind = isUGP ? BATRewardsNotificationKindGrant : BATRewardsNotificationKindGrantAds;
+        
+        [self addNotificationOfKind:notificationKind
+                           userInfo:nil
+                     notificationID:[self notificationIDForPromo:promotion.cppObjPtr]
+                           onlyOnce:YES];
       }
     }
     if (completion) {
-      completion(self.pendingGrants);
+      completion();
+    }
+    for (BATBraveLedgerObserver *observer in [self.observers copy]) {
+      if (observer.promotionsAdded) {
+        observer.promotionsAdded(self.pendingPromotions);
+      }
+      if (observer.finishedPromotionsAdded) {
+        observer.finishedPromotionsAdded(self.finishedPromotions);
+      }
     }
   });
 }
 
-- (void)grantCaptchaForPromotionId:(NSString *)promoID promotionType:(NSString *)promotionType completion:(void (^)(NSString * _Nonnull, NSString * _Nonnull))completion
+- (void)fetchPromotions:(nullable void (^)(NSArray<BATPromotion *> *grants))completion
 {
-  std::vector<std::string> headers;
-  headers.push_back("brave-product:brave-core");
-  headers.push_back("promotion-id:" + std::string(promoID.UTF8String));
-  headers.push_back("promotion-type:" + std::string(promotionType.UTF8String));
-  ledger->GetGrantCaptcha(headers,
-      ^(const std::string &image, const std::string &hint) {
-        dispatch_async(dispatch_get_main_queue(), ^{
-          completion([NSString stringWithUTF8String:image.c_str()],
-                     [NSString stringWithUTF8String:hint.c_str()]);
-        });
-      });
+  ledger->FetchPromotions(^(ledger::Result result, std::vector<ledger::PromotionPtr> promotions) {
+    if (result != ledger::Result::LEDGER_OK) {
+      return;
+    }
+    [self updatePromotions:^{
+      if (completion) {
+        completion(self.pendingPromotions);
+      }
+    }];
+  });
 }
 
-- (void)solveGrantCaptchWithPromotionId:(NSString *)promotionId solution:(NSString *)solution
+- (void)claimPromotion:(NSString *)deviceCheckPublicKey completion:(void (^)(BATResult result, NSString * _Nonnull noonce))completion
 {
-  ledger->SolveGrantCaptcha(std::string(solution.UTF8String),
-                            std::string(promotionId.UTF8String));
+  const auto payload = [NSDictionary dictionaryWithObject:deviceCheckPublicKey forKey:@"publicKey"];
+  const auto jsonData = [NSJSONSerialization dataWithJSONObject:payload options:0 error:nil];
+  if (!jsonData) {
+    NSLog(@"Missing JSON payload while attempting to claim promotion");
```

Comment:
cc @kylehickinson

Reply:
`NSLog` is usually the default in Obj-C and no they don't compile out. I could swap it out to use same logging as rest of Rewards if that's needed but this string in particular isn't a risk

Reply:
If it's already widely used then no worries - though it might be worth picking up at some point, as we'll eventually be super-spamming the console log.

Reply:
Better anyways, I switched them all to the rewards logger